lms/djangoapps/courseware/access.py

```diff
@@ -379,6 +369,14 @@ def can_load():
             else:
                 return has_not_expired
 
+        user_has_active_subscription = is_course_accessible_with_subscription(user, courselike)
```

Comment:
pushed this condition to the last so that if any other existing condition needs to restrict access, it can.